admin/app/components/solidus_admin/ui/table/component.js

```diff
@@ -0,0 +1,51 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["checkbox", "headerCheckbox", "batchToolbar", "scopesToolbar", "tableHeader", "selectedRowsCount"]
+
+  connect() {
+    this.mode = "default"
+
+    this.render()
+  }
+
+  selectRow(event) {
+    if (this.checkboxTargets.some((checkbox) => checkbox.checked)) {
+      this.mode = "batch"
+    } else {
+      this.mode = "default"
+    }
+
+    this.render()
+  }
+
+  selectAllRows(event) {
+    this.mode = event.target.checked ? "batch" : "default"
+    this.checkboxTargets.forEach((checkbox) => (checkbox.checked = event.target.checked))
+
+    this.render()
+  }
+
+  render() {
+    const selectedRows = this.checkboxTargets.filter((checkbox) => checkbox.checked)
+
+    this.batchToolbarTarget.classList.toggle("hidden", this.mode !== "batch")
+    this.scopesToolbarTarget.classList.toggle("hidden", this.mode !== "default")
+    this.tableHeaderTarget.classList.toggle("hidden", this.mode !== "default")
```

Comment:
Nit: thoughts on using the positive version?

```suggestion
    this.batchToolbarTarget.classList.toggle("hidden", this.mode == "default")
    this.scopesToolbarTarget.classList.toggle("hidden", this.mode == "batch")
    this.tableHeaderTarget.classList.toggle("hidden", this.mode == "batch")
```

Reply:
Another nit: should we pick the "hidden" class from a data attribute?